org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java

```diff
@@ -374,4 +381,11 @@ protected void doClean(ToBeBuilt toBeBuilt, IProgressMonitor monitor) throws Cor
 		}
 	}
 	
+	@Override
+	public ISchedulingRule getRule(int kind, Map<String, String> args) {
+		return new MultiRule(Arrays.stream(
```

Comment:
I'm not terribly sure about the injected workspace. `getProject().getWorkspace()` should be sufficient

Reply:
using workspace (root) as a rule will block all other builders (e.g. JDT, PDE, ...) from using the "new" parallel build capabilities as soon as there is a single Xtext project.

Reply:
To be more explicit:
Instead of `workspace.getRoot().getProjects().etc` I'd rather opt for `getProject().getWorkspace().getRoot().getProjects().etc`